### DISCUSSION TITLE
Output to table

### DISCUSSION THREAD

**Discussion by mazzarin** [mmohades/Venmo#63] [Q&A]
Hi, thanks for creating the project! 

I'm having difficulty converting the json result for a transaction log to a csv file with headers. 

Can you recommend a way forward?

**Comment by mmohades** [comment]
Hey, just seeing this. Were you able to do it?
You may be able to do it using `Pandas` or manually using `csv.writer()` and `writerow()`.

**Comment by mazzarin** [comment]
Unfortunately I was not able to get it working very well! I tried a few things but the furthest I got was a csv output where the script would prepend the header onto every cell which ... works but is highly inconvenient.

Here's the hacky version that works:
```
def callback(transactions_list):
 with open("txvenmo.csv", "w") as outfile:
    for transaction in transactions_list:
        outfile.write(str(transaction) + "\n")
```

This worked as described above.

---

I tried the following with pandas (import pandas as pd) but couldn't get it working
```
transactions = client.user.get_user_transactions(<snip>)
with open (transactions) as inputfile:
 df = pd.read_json(inputfile)
 df.to_csv('venmotxnew.csv', encoding='utf-8', index=False)
print ("Done export.")
```

Obviously transactions isn't the correct input file, but I couldn't figure out how best to structure it.

Any advice is appreciated!
